spec/component_guide/component_index_spec.rb

```diff
@@ -1,6 +1,10 @@
 require "rails_helper"
 
 describe "Component guide index" do
+  around do |example|
+    ClimateControl.modify(GOVUK_ENABLE_SEARCH_AUTOCOMPLETE: "true") { example.run }
+  end
```

Comment:
So the reason adding this in didn't make the tests pass is because the suggested js / css is looking for strings in the code that match the [full component name](https://github.com/alphagov/govuk_publishing_components/blob/219710480ba337d04dee0408f9df8e52ff26e83a/app/controllers/govuk_publishing_components/component_guide_controller.rb#L154). So before a variable was used in the component name it would match `govuk_publishing_components/components/search_with_autocomplete` but it can't match on a variable of `govuk_publishing_components/components/#{search_component}` as the analysis is done on the source code not the product of the source code.

I imagine the way to resolve it most in the spirit of the component gem suggestion functionality would be to do the conditional and have separate component calls. 

E.g:

```
<% search_options = {
  name: "keywords",
  inline_label: false,
  label_size: "m",
  label_text: search_text,
  label_custom_class: "gem-c-layout-super-navigation-header__search-label--large-navbar",
  size: "large",
  margin_bottom: 0,
  disable_corrections: true,
} %>
<% if ENV["GOVUK_ENABLE_SEARCH_AUTOCOMPLETE"] == "true" %>
  <%= render "govuk_publishing_components/components/search_with_autcomplete", search_options.merge({
    source_url: [Plek.new.website_root, "/api/search/autocomplete.json"].join,
    source_key: "suggestions"
  }) %>
<% else %>
  <%= render "govuk_publishing_components/components/search", search_options %>
<% endif %>
<% end %>
```

Though another way of course would be to just remove it from the suggested components list - but that'd be accepting this component couldn't be suggested which feels against the spirit of this functionality.